src/Microsoft.Data.SqlClient/tests/Common/LocalAppContextSwitchesHelper.cs

```diff
@@ -0,0 +1,355 @@
+    }
+    public bool MakeReadAsyncBlocking
+    {
+        get => (bool)_makeReadAsyncBlockingProperty.GetValue(null);
+    }
+    public bool UseMinimumLoginTimeout
+    {
+        get => (bool)_useMinimumLoginTimeoutProperty.GetValue(null);
+    }
+    public bool LegacyVarTimeZeroScaleBehaviour
+    {
+        get => (bool)_legacyVarTimeZeroScaleBehaviourProperty.GetValue(null);
+    }
+    public bool UseCompatibilityProcessSni
+    {
+        get => (bool)_useCompatProcessSniProperty.GetValue(null);
+    }
+    public bool UseCompatibilityAsyncBehaviour
+    {
+        get => (bool)_useCompatAsyncBehaviourProperty.GetValue(null);
+    }
+#if NETFRAMEWORK
+    public bool DisableTNIRByDefault
+    {
+        get => (bool)_disableTNIRByDefaultProperty.GetValue(null);
+    }
+#endif
+
+    // These properties get or set the like-named underlying switch field value.
+    //
+    // They all fail the test if the value cannot be retrieved or set.
+
+    public Tristate LegacyRowVersionNullBehaviorField
+    {
+        get => GetValue(_legacyRowVersionNullBehaviorField);
+        set => SetValue(_legacyRowVersionNullBehaviorField, value);
+    }
+
+    public Tristate SuppressInsecureTLSWarningField
+    {
+        get => GetValue(_suppressInsecureTLSWarningField);
+        set => SetValue(_suppressInsecureTLSWarningField, value);
+    }
+
+    public Tristate MakeReadAsyncBlockingField
+    {
+        get => GetValue(_makeReadAsyncBlockingField);
+        set => SetValue(_makeReadAsyncBlockingField, value);
+    }
+
+    public Tristate UseMinimumLoginTimeoutField
+    {
+        get => GetValue(_useMinimumLoginTimeoutField);
+        set => SetValue(_useMinimumLoginTimeoutField, value);
+    }
+
+    public Tristate LegacyVarTimeZeroScaleBehaviourField
+    {
+        get => GetValue(_legacyVarTimeZeroScaleBehaviourField);
+        set => SetValue(_legacyVarTimeZeroScaleBehaviourField, value);
+    }
+
+    public Tristate UseCompatProcessSniField
+    {
+        get => GetValue(_useCompatProcessSniField);
+        set => SetValue(_useCompatProcessSniField, value);
+    }
+
+    public Tristate UseCompatAsyncBehaviourField
+    {
+        get => GetValue(_useCompatAsyncBehaviourField);
+        set => SetValue(_useCompatAsyncBehaviourField, value);
+    }
+
+#if NETFRAMEWORK
+    public Tristate DisableTNIRByDefaultField
+    {
+        get => GetValue(_disableTNIRByDefaultField);
+        set => SetValue(_disableTNIRByDefaultField, value);
+    }
+#endif
+
+    #endregion
+
+    #region Private Helpers
+
+    private static Tristate GetValue(FieldInfo field)
+    {
+        var value = field.GetValue(null);
+        if (value is null)
+        {
+            Assert.Fail($"Field {field.Name} has a null value.");
+        }
+
+        return (Tristate)value;
+    }
+
+    private static void SetValue(FieldInfo field, Tristate value)
+    {
+        field.SetValue(null, (byte)value);
+    }
+
+    #endregion
+
+    #region Private Members
```

Comment:
Whoa whoa whoa ... private members should be at the top of the class!

Reply:
I guess it's ultimately a "this is what we've always done" scenario, and probably comes from the olden days when you had to declare what was being used in a scope at the beginning of the scope so the compiler could determine how much memory to allocate. To me it makes more sense to define what the class is storing before defining what it does. But that might be me searching for a rationale for why we should keep doing it now. Circular logic, I guess.